lib/galaxy/webapps/galaxy/api/authenticate.py

```diff
@@ -92,7 +92,15 @@ def _decode_baseauth(self, encoded_str):
 
         :raises: HTTPBadRequest
         """
-        split = encoded_str.strip().split(' ')
+        try:
+            split = encoded_str.strip().split(' ')
+        except AttributeError:
+            if not encoded_str:
+                raise exceptions.MessageException(
+                    'Authentication is missing')
+            else:
+                raise exceptions.MessageException(
+                    'Failed to authenticate')
```

Comment:
This is more readable if we don't wrap it so short, right?  We don't strictly adhere to 80 in the Galaxy codebase.

```suggestion
                raise exceptions.MessageException('Authentication is missing')
            else:
                raise exceptions.MessageException('Failed to authenticate')
```